keps/sig-node/727-resource-metrics-endpoint/README.md

```diff
@@ -106,48 +169,243 @@ Labels are named in accordance with the [kubernetes instrumentation guidelines](
+
+Beta:
+
+- [X] Modify the metrics server to consume the kubelet resource metrics endpoint 3 releases after it is added to the kubelet
+
+GA:
+
+- [X] Add [node-e2e test](https://github.com/kubernetes/kubernetes/pull/116897/files#diff-3859a7587ac4b3d1e162a2360b1fd2d3e88d4589be9b0bf19029fa7489294796R59-R70) to the node conformance tests
+
+### Upgrade / Downgrade Strategy
+
+The kubelet can be upgraded or downgraded normally with respect to this feature. Users of the metrics endpoint, such as the metrics server, should use other kubelet metrics endpoints (such as the summary api) before downgrading.
+
+### Version Skew Strategy
+
+This feature affects only the kubelet - in that it will expose the resource metrics for kubelet in a new endpoint, so there is no issue with version skew with other components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
+  - Components depending on the feature gate:
+- [x] Other
+  - Describe the mechanism: This feature exposes the /metrics/resource endpoint for kubelet, with all metrics annotated as STABLE 
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No
+
+###### Does enabling the feature change any default behavior?
+
+It will expose the /metrics/resource endpoint for kubelet by default
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+No, this feature can not be disabled once it has been enabled since we do not have a feature flag for this. To rollback, one will have to downgrade the kubernetes version.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+/metrics/resource endpoint for kubelet will become available
+
+###### Are there any tests for feature enablement/disablement?
+
+Since there is no feature gate involved for this, there are no feature enablement/disablement test
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollback can impact running workloads if clients, such as the metrics server, are relying on metrics provided by the endpoint. The rollback could break cluster functions, such as HPA, if the metrics were no longer available.
+
+###### What specific metrics should inform a rollback?
+
+The following metrics exposed by /kubelet/resource endpoint could be used:
+- node_memory_working_set_bytes
+- pod_memory_working_set_bytes
+
+We could compute node_memory_working_set_bytes - sum(pod_memory_working_set_bytes) to know if there's a memory leak. 
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No
```

Comment:
Given it's "always enabled" for last 14 releases, it's too late now.

Let's maybe make it clear, sth like

"It was enabled (with no way to disable) since 1.14."

Reply:
Done